src/core/Akka.Streams/Dsl/Sink.cs

```diff
@@ -325,7 +325,7 @@ public static class Sink
         /// <param name="action">Async function delegate to be executed on all elements</param>
         /// <returns>TBD</returns>
         public static Sink<TIn, Task<Done>> ForEachAsync<TIn>(int parallelism, Func<TIn, Task> action) => Flow.Create<TIn>()
-            .SelectAsync(parallelism, async input =>
+            .SelectAsyncUnordered(parallelism, async input =>
```

Comment:
This will break the intent of `ForEachAsync`, we should preserve the behavior of our flow stream to be idiomatic to C#.
In this case a `foreach` should process the stream in order, which is why we need to use `SelectAsync`

Reply:
Actually, this replaces `ForEachParallel` which already utilizes `SelectAsyncUnordered`. The change in behavior was introduced by the current implementation of `ForEachAsync`.

Keep in mind that this operator is similar to `Parallel.ForEach` or `Parallel.ForEachAsync` in C#, which *also* does not guarantee the order of execution.

Reply:
OK, makes sense